site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param VRLayerInit layers: array of layers which will be rendered to.
+	:param int layerCount: number of layers in `layers`.
+	:param em_vr_arg_callback_func callback: optional function that will be called when the requst has succeeded.
+	:param void* userData: optional data to pass to the callback when the request succeeds. Is not modified by the API.
+	:rtype: |display-function-return-doc|
+
+.. c:function:: int emscripten_vr_exit_present(VRDisplayHandle handle)
+
+	Request present exit.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:rtype: |display-function-return-doc|
+
+.. _defines-vr-api:
+
+Defines
+-------
+
+.. c:macro:: VR_EYE_LEFT
+	VR_EYE_RIGHT
+
+	Eye values for use with :c:func:`emscripten_vr_get_eye_parameters`.
+
+.. _vr-pose-defines-vr-api:
+
+.. c:macro:: VR_POSE_POSITION
+	VR_POSE_LINEAR_VELOCITY
+	VR_POSE_LINEAR_ACCELERATION
+	VR_POSE_ORIENTATION
+	VR_POSE_ANGULAR_VELOCITY
+	VR_POSE_ANGULAR_ACCELERATION
+
+	Flags which describe which properties of a :c:type:`VRPose` are valid.
+
+.. c:macro:: VR_LAYER_DEFAULT_LEFT_BOUNDS
+	VR_LAYER_DEFAULT_RIGHT_BOUNDS
+
+	Default values to pass to :c:type:`VRLayerInit`.
+
+.. _types-vr-api:
+
+Types
+-----
+
+.. c:type:: VRDisplayCapabilities
+
+	Structure passed to :c:func:`emscripten_vr_get_display_capabilities`, maps to the WebVR `VRDisplayCapabilities <https://w3c.github.io/webvr/spec/1.1/#interface-vrdisplaycapabilities>`__ interface.
+
+	.. c:member:: int32_t hasPosition
+
+	.. c:member:: int32_t hasExternalDisplay
+
+	.. c:member:: int32_t canPresent
+
+	.. c:member:: unsigned long maxLayers
+
+
+.. c:type:: VRLayerInit
+
+	Structure passed to :c:func:`emscripten_vr_request_present`, maps to the WebVR `VRLayerInit <https://w3c.github.io/webvr/spec/1.1/#interface-vrlayerinit>`__ interface.
+
+	.. c:member:: const char* source
+
+		Id of the source canvas which will be used to present to the VR display.
+
+		`0 (NULL)` is used to refer to ``Module.canvas``.
+
+	.. c:member:: float[4] leftBounds
+
+		Texture bounds of the left eye on the target canvas. Initialize with :c:macro:`VR_LAYER_DEFAULT_LEFT_BOUNDS` for default.
+
+	.. c:member:: float[4] rightBounds
+
+		Texture bounds of the right eye on the target canvas. Initialize with :c:macro:`VR_LAYER_DEFAULT_RIGHT_BOUNDS` for default.
+
+
+.. c:type:: VRPose
+
+	Substructure of :c:type:`VRFrameData`, maps to the WebVR
+	`VRPose <https://w3c.github.io/webvr/spec/1.1/#interface-vrpose>`__ interface.
+
+	VR Displays do not necessarily report all of the pose values (mobile VR devices usually
+	only report orientation, but not position for example). To check which values are valid,
+	the :c:member:`poseFlags <poseFlags>` member provides a bitmask of
+	:ref:`VR_POSE_* <vr-pose-defines-vr-api>` which has a bit set for every valid value.
+
+	.. c:member:: VRVector3 position
+
+		Position, valid only if ``poseFlags & VR_POSE_POSITION == 0``.
```

Comment:
Should this not be the other way around? In above it says `which has a bit set for every valid value`, but here it is testing the other way around, i.e. one'd expect

```
, valid only if ``(poseFlags & VR_POSE_POSITION) != 0``.
```

(also I think it will be good to have the extra parentheses here on the docs as well, since otherwise LLVM will issue a compiler warning if one copy pastes the example code as-is)

Reply:
> Should this not be the other way around?

Yes, definitely, sorry about that!